db/sqlinterfaces.c

```diff
@@ -1436,71 +1436,61 @@ void sql_dump_hist_statements(void)
+    char addr[64];
+
+    int fd = sbuf2fileno(clnt->sb);
+    rc = getpeername(fd, (struct sockaddr *)&peeraddr, &len);
+    if (rc)
+        snprintf(addr, sizeof(addr), "<unknown>");
+    else {
+        if (inet_ntop(peeraddr.sin_family, &peeraddr.sin_addr, addr,
+                      sizeof(addr)) == NULL)
             snprintf(addr, sizeof(addr), "<unknown>");
-        else {
-            if (inet_ntop(peeraddr.sin_family, &peeraddr.sin_addr, addr,
-                          sizeof(addr)) == NULL)
-                snprintf(addr, sizeof(addr), "<unknown>");
-        }
+    }
 
-        if (clnt->limits.maxcost_warn &&
-            (thd->cost > clnt->limits.maxcost_warn)) {
-            logmsg(LOGMSG_USER, 
-                   "[%s] warning: query exceeded cost threshold (%f >= %f): %s\n",
-                   addr, thd->cost, clnt->limits.maxcost_warn, clnt->sql);
-        }
-        if (clnt->limits.tablescans_warn && thd->had_tablescans) {
-            logmsg(LOGMSG_USER, 
-                   "[%s] warning: query had a table scan: %s\n", addr,
-                   clnt->sql);
-        }
-        if (clnt->limits.temptables_warn && thd->had_temptables) {
-            logmsg(LOGMSG_USER, 
-                   "[%s] warning: query created a temporary table: %s\n", addr,
-                   clnt->sql);
-        }
-        if (clnt->osql.rqid != 0 && clnt->osql.rqid != OSQL_RQID_USE_UUID)
-            reqlog_set_rqid(logger, &clnt->osql.rqid, sizeof(clnt->osql.rqid));
-        else {
-            /* have an "id_set" instead? */
-            if (!comdb2uuid_is_zero(clnt->osql.uuid))
-                reqlog_set_rqid(logger, clnt->osql.uuid, sizeof(uuid_t));
-        }
+    if (clnt->limits.maxcost_warn && (thd->cost > clnt->limits.maxcost_warn)) {
+        logmsg(LOGMSG_USER,
+               "[%s] warning: query exceeded cost threshold (%f >= %f): %s\n",
+               addr, thd->cost, clnt->limits.maxcost_warn, clnt->sql);
+    }
+    if (clnt->limits.tablescans_warn && thd->had_tablescans) {
+        logmsg(LOGMSG_USER, "[%s] warning: query had a table scan: %s\n", addr,
+               clnt->sql);
+    }
+    if (clnt->limits.temptables_warn && thd->had_temptables) {
+        logmsg(LOGMSG_USER,
+               "[%s] warning: query created a temporary table: %s\n", addr,
+               clnt->sql);
     }
 
+    unsigned long long rqid = clnt->osql.rqid;
+    if (rqid != 0 && rqid != OSQL_RQID_USE_UUID)
+        reqlog_set_rqid(logger, &rqid, sizeof(rqid));
+    else {
+        /* have an "id_set" instead? */
+        if (!comdb2uuid_is_zero(clnt->osql.uuid))
+            reqlog_set_rqid(logger, clnt->osql.uuid, sizeof(uuid_t));
+    }
+
+    LISTC_T(struct sql_hist) lst;
     listc_init(&lst, offsetof(struct sql_hist, lnk));
 
-    h = calloc(1, sizeof(struct sql_hist));
-    if (thd->sqlclntstate && thd->sqlclntstate->sql)
-        h->sql = strdup(thd->sqlclntstate->sql);
+    struct sql_hist *h = calloc(1, sizeof(struct sql_hist));
```

Comment:
Any time  you feel the need to cluster variables and associated code together, there is a chance to put everything in its own encapsulating function, just saying

Reply:
I embrace C99 whole heartedly :-p